src/main/java/hwr/oop/group4/todo/ui/ProjectUi.java

```diff
@@ -0,0 +1,211 @@
+                        new EntryArgument("id <id>", "ID of the project."))),
+                new Entry("edit",   "Edit the attributes of a project.", List.of(
+                        new EntryArgument("id <id>", "ID of the project to be edited."),
+                        new EntryArgument("name <name>", "Change the name of the project."),
+                        new EntryArgument("desc <desc>", "Change the description of the project."),
+                        new EntryArgument("begin", "Change the beginning of the project."),
+                        new EntryArgument("end", "Change the end of the project"),
+                        new EntryArgument("addTag <tag>", "Add a new tag."),
+                        new EntryArgument("removeTag <tag>", "Remove a tag."))),
+                new Entry("remove", "Remove a project.", List.of(
+                        new EntryArgument("id <id>", "ID of the project to be removed."))),
+                new Entry("back",   "Returns to the previous menu.")
+        ));
+        consoleController.output(menu.toString());
+
+        AtomicBoolean shouldReturn = new AtomicBoolean(false);
+        while (!shouldReturn.get()) {
+            consoleController.inputOptions(List.of("projects"), List.of(
+                    new Command("list",   this::listProjects),
+                    new Command("new",    this::newProject),
+                    new Command("tasks",  args -> {}),
+                    new Command("edit",   this::editProject),
+                    new Command("remove", this::removeProject),
+                    new Command("back",   args -> shouldReturn.set(true))
+            ), new Command("wrongInput", args -> {}));
+        }
+    }
+
+    private void listProjects(Collection<CommandArgument<String>> args) {
+        final List<Project> projects = todoList.getProjects();
+        final int idColumnLength = Math.max((int) Math.ceil(Math.log10(projects.size()) - 2), 2);
+        final Table projectTable = new Table(List.of(
+                new ColumnConfig("ID", idColumnLength),
+                new ColumnConfig("Name", 15),
+                new ColumnConfig("Description", 30),
+                new ColumnConfig("Tags", 10),
+                new ColumnConfig("Begin", 6),
+                new ColumnConfig("End", 6)
+        ));
+
+        for (int i = 0; i < projects.size(); i++) {
+            final Project project = projects.get(i);
+            projectTable.addRow(
+                    String.valueOf(i),
+                    project.getName(),
+                    project.getDescription(),
+                    concatTagsToString(project.getTags()),
+                    project.getBegin().format(DateTimeFormatter.ofPattern("dd.MM.")),
+                    project.getEnd().format(DateTimeFormatter.ofPattern("dd.MM."))
+            );
+        }
+
+        consoleController.output(projectTable.toString());
+    }
+
+    private String concatTagsToString(Collection<Tag> tags) {
```

Comment:
can be moved somewhere else to make the class shorter

Reply:
Same for getId(). Could possible be moved to ConsoleController, which is already 149 lines.

**Suggestion**. Split ConsoleController into three classes:
- Input
- Output
- Parse/Additionals/Misc

Reply:
Moved getId(), concatTagsToString(), getStringParameter(), parseDate() to ConsoleHelper for now to reduce both ProjectUi and ConsoleController.